src/dune_rules/resolve.ml

```diff
@@ -41,6 +41,18 @@ let to_result x = x
 
 let of_error x = Error x
 
+let raise_error ({ exn; _ } : error) = raise exn
+
+let push_frames { stack_frames; exn = _ } =
+  let rec loop = function
+    | [] -> Memo.return ()
+    | x :: rest ->
+      Memo.push_stack_frame
+        ~human_readable_description:(fun () -> Lazy.force x)
+        (fun () -> loop rest)
+  in
+  loop stack_frames
```

Comment:
Can you change error_to_memo to use this function?

Reply:
I tried to do that but a single frame went missing in a bunch of tests

Reply:
I pushed the change. Let's see what CI says.